src/main/scala/tech/sourced/gemini/ReportSparkApp.scala

```diff
@@ -1,34 +1,46 @@
 package tech.sourced.gemini
 
+import com.datastax.driver.core.Cluster
+
 object ReportSparkApp extends App {
   def printUsage(): Unit = {
-    println("Usage: ./report <repository>")
+    println("Usage: ./report [--detailed]")
     println("")
-    println("Finds duplicated files among hashed repositories")
-    println("  <repository> - repository url, example: github.com/src-d/go-git.git")
+    println("Finds duplicate files among hashed repositories")
+    println("  --detailed to see file paths of each duplicated file")
     System.exit(2)
   }
 
-  if (args.length <= 0) {
-    printUsage()
+  def print(report: Iterable[Any], detailed: Boolean): Unit = {
+    if (report.isEmpty) {
+      println(s"No duplicates found.")
+    } else if (detailed) {
```

Comment:
is early return common practice in Scala?

Reply:
Usually it is not.
If you have concrete suggestions on restructuring the code here - please feel free to share examples as well.

Reply:
Regular early return as in go or js or many other languages:
```scala
def print(report: Iterable[Any], detailed: Boolean): Unit = {
    if (report.isEmpty) {
        println(s"No duplicates found.")
        return
    }

    if (detailed) {
        report.foreach { item =>
        val duplicateFiles = item.asInstanceOf[Iterable[RepoFile]]
        val count = duplicateFiles.count(_ => true)
        println(s"$count duplicates:\n\t" + (duplicateFiles mkString "\n\t") + "\n")
        return
    } 

    println(s"Duplicates found:\n\t" + (report mkString "\n\t"))
}
```

Reply:
Yes, and that is not common practice and generally shall be avoided in Scala. 
Here is more context https://tpolecat.github.io/2014/05/09/return.html

Reply:
Oh I see why now. Thanks!

Actually it makes sense, you always can use pattern matching.

```scala
sealed abstract class Result {
    def size(): Int
}
case class ResultFiles(vs: List[RepoFile]) extends Result {
    override def size() = vs.size
}
case class ResultBlobs(vs: List[DuplicateBlobHash]) extends Result {
    override def size() = vs.size
}

def print(report: Result): Unit = {
    report match {
        case empty if empty.size == 0 => println(s"No duplicates found.")
        case ResultFiles(vs) => println(vs.size+" duplicates:\n\t" + (vs mkString "\n\t") + "\n")
        case ResultBlobs(vs) => println(s"Duplicates found:\n\t" + (vs mkString "\n\t"))
    }
}
```

Reply:
not need to rewrite though. Just example of cool scala syntax.